File: backend/src/schemas/cars.py
"""Car request/response models."""
from __future__ import annotations

from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class CarCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=120)
    year: int | None = None
    color_label: str | None = None
    color: str = "stone"
    notes: str | None = None


class CarUpdateRequest(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=120)
    year: int | None = None
    color_label: str | None = None
    color: str | None = None
    notes: str | None = None


class CarResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    family_id: UUID
    name: str
    year: int | None = None
    color_label: str | None = None
    color: str
    notes: str | None = None
    status: Literal["active", "inactive"]
    created_at: datetime

File: frontend/src/lib/api/cars.ts
import { http, jsonBody } from "./http";
import type { MemberColor } from "@/components/fridge/types";

export type CarStatus = "active" | "inactive";

export interface CarResponse {
  id: string;
  family_id: string;
  name: string;
  year: number | null;
  color_label: string | null;
  color: MemberColor;
  notes: string | null;
  status: CarStatus;
  created_at: string;
}

export interface CarCreateRequest {
  name: string;
  year?: number | null;
  color_label?: string | null;
  color: MemberColor;
  notes?: string | null;
}

export type CarUpdateRequest = Partial<CarCreateRequest>;

export type CarStatusFilter = "active" | "inactive" | "all";

export const carsApi = {
  list: (status: CarStatusFilter = "active") =>
    http<CarResponse[]>(`/api/cars?status=${status}`),
  create: (body: CarCreateRequest) =>
    http<CarResponse>("/api/cars", { method: "POST", body: jsonBody(body) }),
  update: (id: string, body: CarUpdateRequest) =>
    http<CarResponse>(`/api/cars/${id}`, {
      method: "PATCH",
      body: jsonBody(body),
    }),
  setActive: (id: string) =>
    http<CarResponse>(`/api/cars/${id}/set-active`, { method: "POST" }),
  setInactive: (id: string) =>
    http<CarResponse>(`/api/cars/${id}/set-inactive`, { method: "POST" }),
  delete: (id: string) =>
    http<void>(`/api/cars/${id}`, { method: "DELETE" }),
};
